core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -284,6 +384,15 @@ constructor(
                 .tryApplyImageFormatConstraints()
                 .tryApplyFrameRateConstraints()
                 .tryApplyStabilizationConstraints()
+        if (isDebugMode && Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            val cameraProperties = getAllCamerasPropertiesJSONArray().toString()
+            val file = File(
+                Environment.getExternalStoragePublicDirectory(DIRECTORY_DOCUMENTS),
+                "JCACameraProperties"
+            )
+            writeFileExternalStorage(file, cameraProperties)
```

Comment:
This should be done on a background thread so we're not blocking the main thread with disk IO. You can use `withContext(defaultDispatcher) { ...  }` to run this.

Reply:
Is that any reason you recommend for using a defaultDispatcher, not an IO related dispatcher?

Reply:
You're right, an IO dispatcher would be better. I only recommended `defaultDispatcher` because we already have it available in this class. We should inject an IO dispatcher to use instead.